### PR TITLE
Fix #8755: Uncaught exception Ltac_plugin.Taccoerce.CannotCoerceTo.

### LINKED ISSUE #8755: Uncaught exception Ltac_plugin.Taccoerce.CannotCoerceTo
#### Version

8.8.2

#### Operating system

Red Hat Enterprise Linux Workstation 7.4

#### Description of the problem

The following nonsense causes an anomaly.

```
Tactic Notation "recall" ident (i) :=
  let t := context i [tt] in pose proof t as i.

Inductive Yes : Type := yes.

Example certainly : Yes.
Proof. recall yes. auto. Qed.
```

```
Anomaly "Uncaught exception Ltac_plugin.Taccoerce.CannotCoerceTo("a term context")."
```

### PR DESCRIPTION
**Pull request by ppedrot**
Fixes #8755.

### CODE REVIEW

**Comment by ppedrot** [comment]
Should probably be assigned to @herbelin.

File: plugins/ltac/tacinterp.ml [hunk 1]
```diff
@@ -687,7 +687,7 @@ let interp_may_eval f ist env sigma = function
   | ConstrContext ({loc;v=s},c) ->
       (try
 	let (sigma,ic) = f ist env sigma c in
-	let ctxt = coerce_to_constr_context (Id.Map.find s ist.lfun) in
+        let ctxt = try_interp_ltac_var coerce_to_constr_context ist (Some (env, sigma)) (make ?loc s) in
```
Comment (herbelin) [inline plugins/ltac/tacinterp.ml]: At first view, an eta-contraction of the `{loc;v=s}` is possible.

Otherwise assigning.

**Review by herbelin — APPROVED** [review]
Thanks, assigning.

File: plugins/ltac/tacinterp.ml (hunk 1 above)
Comment (herbelin) [reply plugins/ltac/tacinterp.ml]: Pedrot, are you ok for this mini-simplification (I can then do it at the time of merging if you want)?

Comment (JasonGross) [reply plugins/ltac/tacinterp.ml]: @herbelin If you eta-contract `{loc;v=s}`, how should `user_err ?loc ~hdr:"interp_may_eval" (str "Unbound context identifier" ++ Id.print s ++ str".")` be updated?

Comment (JasonGross) [reply plugins/ltac/tacinterp.ml]: We still need `loc` for the `?loc` argument to `user_err` anyway, right?

**Comment by ppedrot** [comment]
@herbelin As supported by @JasonGross, I think this is ready to merge.

**Comment by herbelin** [comment]
> @herbelin As supported by @JasonGross, I think this is ready to merge.

OK. I failed to see the different types. Shall merge in half an hour.

**Pull request merged**